services/121-service/src/payments/fsp-integration/nedbank/nedbank-api.service.spec.ts

```diff
@@ -0,0 +1,273 @@
+
+      await expect(
+        service.createOrder({
+          transferAmount: amount,
+          phoneNumber: registrationNedbank.phoneNumber,
+          orderCreateReference,
+        }),
+      ).rejects.toThrow(NedbankError);
+
+      // TODO: Not sure if this is the best/prettiest syntax to test the content of the error but it works
+      let errorOnCreateOrder: NedbankError | undefined;
+      try {
+        await service.createOrder({
+          transferAmount: amount,
+          phoneNumber: registrationNedbank.phoneNumber,
+          orderCreateReference,
+        });
+      } catch (error) {
+        errorOnCreateOrder = error;
+      }
+      expect(errorOnCreateOrder).toMatchSnapshot();
+    });
+
+    it('should throw an error if httpsAgent is not defined', async () => {
+      service.httpsAgent = undefined;
+
+      // TODO: Not sure if this is the best/prettiest syntax to test the content of the error but it works
+      let errorOnCreateOrder: NedbankError | undefined;
+      try {
+        await service.createOrder({
+          transferAmount: amount,
+          phoneNumber: registrationNedbank.phoneNumber,
+          orderCreateReference,
+        });
+      } catch (error) {
+        errorOnCreateOrder = error;
+      }
+      expect(errorOnCreateOrder).toMatchSnapshot();
+    });
+  });
+
+  describe('getOrder', () => {
+    it('should get an order successfully', async () => {
+      const response: AxiosResponse<GetOrderResponseNedbankDto> = {
+        data: {
+          Data: {
+            OrderId: '',
+            Transactions: {
+              Voucher: {
+                Code: '',
+                Status: NedbankVoucherStatus.REDEEMED,
+                Redeem: {
+                  Redeemable: true,
+                  Redeemed: false,
+                  RedeemedOn: '',
+                  RedeemedAt: '',
+                },
+                Refund: {
+                  Refundable: true,
+                  Refunded: false,
+                  RefundedOn: '',
+                },
+                Pin: '',
+              },
+              PaymentReferenceNumber: '',
+              OrderCreateReference: orderCreateReference,
+              OrderDateTime: '',
+              OrderExpiry: '',
+            },
+          },
+          Links: {
+            Self: '',
+          },
+          Meta: {},
+        },
+        status: 200,
+        statusText: 'OK',
+        headers: {},
+        config: {},
+      };
+      jest.spyOn(httpService, 'request').mockResolvedValue(response);
+
+      const result =
+        await service.getOrderByOrderCreateReference(orderCreateReference);
+      expect(result).toEqual(response.data.Data.Transactions.Voucher.Status);
+
+      const requestCallArgs = (httpService.request as jest.Mock).mock.calls[0];
+
+      // Check the URL
+      expect(requestCallArgs[0].url).toContain(`orders/references/`);
+
+      // Check the method
+      expect(requestCallArgs[0].method).toBe('GET');
```

Comment:
More of an integration test than unit test?

Reply:
Comment is outdated and I parked this in a ##TODO other discussion